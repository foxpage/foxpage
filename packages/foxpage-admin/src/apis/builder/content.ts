import {
  BaseResponse,
  ContentFetchParams,
  ContentPublishedRes,
  ContentPublishParams,
  ContentSavedRes,
  ContentSaveParams,
  PageContent,
} from '@/types/index';
import FoxPageApi from '@/utils/api-agent';

export const fetchPageBuildVersion = (params: ContentFetchParams): Promise<PageContent> =>
  new Promise((resolve) => {
    FoxPageApi.get('/pages/build-versions', params, (rs: PageContent) => {
      resolve(rs);
    });
  });
export const fetchPageLiveVersion = (params: ContentFetchParams): Promise<PageContent> =>
  new Promise((resolve) => {
    FoxPageApi.get('/pages/live-version', params, (rs: PageContent) => {
      resolve(rs);
    });
  });

export const fetchTemplateBuildVersion = (params: ContentFetchParams): Promise<PageContent> =>
  new Promise((resolve) => {
    FoxPageApi.get('/templates/build-versions', params, (rs: PageContent) => {
      resolve(rs);
    });
  });

export const fetchPageDsl = (params: ContentFetchParams) =>
  new Promise((resolve) => {
    FoxPageApi.post('/pages/draft-infos', params, (rs: BaseResponse) => {
      resolve(rs);
    });
  });

export const fetchTemplateDsl = (params: ContentFetchParams) =>
  new Promise((resolve) => {
    FoxPageApi.post('/templates/draft-infos', params, (rs: BaseResponse) => {
      resolve(rs);
    });
  });

export const savePageContent = (params: ContentSaveParams): Promise<ContentSavedRes> =>
  new Promise((resolve) => {
    FoxPageApi.put('/pages/versions', params, (rs: ContentSavedRes) => {
      resolve(rs);
    });
  });

export const saveTemplateContent = (params: ContentSaveParams): Promise<ContentSavedRes> =>
  new Promise((resolve) => {
    FoxPageApi.put('/templates/versions', params, (rs: ContentSavedRes) => {
      resolve(rs);
    });
  });

export const publishPage = (params: ContentPublishParams) =>
  new Promise((resolve) => {
    FoxPageApi.put('/pages/publish', params, (rs: ContentPublishedRes) => {
      resolve(rs);
    });
  });

export const publishTemplate = (params: ContentPublishParams) =>
  new Promise((resolve) => {
    FoxPageApi.put('/templates/publish', params, (rs: ContentPublishedRes) => {
      resolve(rs);
    });
  });
