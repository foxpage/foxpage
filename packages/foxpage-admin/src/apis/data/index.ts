import { DataBaseQueryParams, ResponseBody } from '@/types/index';
import FoxPageApi from '@/utils/api-agent';

export const fetchCollectionList = (): Promise<ResponseBody> =>
  new Promise((resolve) => {
    FoxPageApi.get('/databases/collections', {}, (rs: ResponseBody) => {
      resolve(rs);
    });
  });

export const queryDataBase = (params: DataBaseQueryParams): Promise<ResponseBody> =>
  new Promise((resolve) => {
    FoxPageApi.post('/databases/query', params, (rs: ResponseBody) => {
      resolve(rs);
    });
  });
