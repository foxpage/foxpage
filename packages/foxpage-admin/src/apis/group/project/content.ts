import FoxpageApi from '../../../utils/foxpage-api-sdk';

export const fetchPageContents = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.get('/pages/content-searchs', params, (rs: any) => {
      resolve(rs);
    });
  });
export const fetchTemplateContents = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.get('/templates/content-searchs', params, (rs: any) => {
      resolve(rs);
    });
  });

export const addPageContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.post('/pages/contents', params, (rs: any) => {
      resolve(rs);
    });
  });
export const addTemplateContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.post('/templates/contents', params, (rs: any) => {
      resolve(rs);
    });
  });
export const updatePageContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.put('/pages/contents', params, (rs: any) => {
      resolve(rs);
    });
  });
export const updateTemplateContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.put('/templates/contents', params, (rs: any) => {
      resolve(rs);
    });
  });
export const deletePageContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.put('/pages/content-status', params, (rs: any) => {
      resolve(rs);
    });
  });
export const deleteTemplateContent = (params: any) =>
  new Promise((resolve) => {
    FoxpageApi.put('/templates/content-status', params, (rs: any) => {
      resolve(rs);
    });
  });
