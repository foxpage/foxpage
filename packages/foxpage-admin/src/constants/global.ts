export enum FileTypeEnum {
  component = 'component',
  page = 'page',
  template = 'template',
  condition = 'condition',
  variable = 'variable',
  function = 'function',
  package = 'package',
}

export const ZH_CN_STRING = 'zh-cn';
export const EN_STRING = 'en';
