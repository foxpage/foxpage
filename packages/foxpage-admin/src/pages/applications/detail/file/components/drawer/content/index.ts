export { default as ContentEditDrawer } from './EditDrawer';
