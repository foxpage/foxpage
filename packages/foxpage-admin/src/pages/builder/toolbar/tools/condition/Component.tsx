import React, { useCallback, useContext, useEffect, useState } from 'react';
import { connect } from 'react-redux';

import { DeleteOutlined, EditOutlined, EyeOutlined, PlusOutlined } from '@ant-design/icons';
import { Button, Divider, Drawer, message, Popconfirm, Spin, Table } from 'antd';
import { DrawerProps } from 'antd/lib/drawer';

import * as ACTIONS from '@/actions/builder/condition';
import { Pagination } from '@/components/common/index';
import { ScopeEnum } from '@/constants/scope';
import RightCloseIcon from '@/pages/builder/component/close';
import ScopeSelect from '@/pages/components/common/ScopeSelect';
import GlobalContext from '@/pages/GlobalContext';
import { ConditionItem } from '@/types/application/condition';

import BuilderContext from '../../../BuilderContext';
import { DrawerHeader, Toolbar } from '../common/CommonStyles';

import EditDrawer from './EditDrawer';

const PAGE_SIZE = 10;

const TYPE = ['and', 'or'];

const mapStateToProps = (state: any) => ({
  applicationId: state.builder.page.applicationId,
  folderId: state.builder.page.folderId,
  pageNum: state.builder.condition.pageNum,
  total: state.builder.condition.total,
  fetching: state.builder.condition.fetching,
  list: state.builder.condition.list,
});

const mapDispatchToProps = {
  changeOffset: ACTIONS.changeOffset,
  openDrawer: ACTIONS.updateConditionDrawerVisible,
  fetchList: ACTIONS.fetchList,
  deleteCondition: ACTIONS.deleteCondition,
};

interface DProps extends Omit<DrawerProps, 'onClose'> {
  onClose?: () => void;
}

type IProps = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps & DProps;

function Component(props: IProps) {
  const {
    applicationId,
    folderId,
    onClose,
    pageNum,
    total,
    fetching,
    list,
    changeOffset,
    openDrawer,
    fetchList,
    deleteCondition,
  } = props;
  const { container } = useContext(BuilderContext);
  const [group, setGroup] = useState<ScopeEnum>(ScopeEnum.project);
  const { locale } = useContext(GlobalContext);
  const { global, condition: conditionI18n } = locale.business;

  useEffect(() => {
    if (applicationId) {
      const params =
        group === ScopeEnum.project && !!folderId
          ? {
              applicationId,
              folderId,
              page: pageNum,
              size: PAGE_SIZE,
            }
          : {
              applicationId,
              type: 'live',
              page: pageNum,
              size: PAGE_SIZE,
            };

      fetchList(params);
    }
  }, [applicationId, folderId, group, pageNum, fetchList]);

  const handleConditionDelete = useCallback(
    (id) => {
      if (id && applicationId) {
        deleteCondition({
          applicationId,
          id,
          status: true,
        });
      } else {
        message.warning(conditionI18n.deleteFailMsg);
      }
    },
    [applicationId, folderId, deleteCondition],
  );

  const columns = [
    {
      title: global.nameLabel,
      dataIndex: 'name',
      key: 'name',
    },
    {
      title: global.type,
      dataIndex: 'type',
      key: 'type',
      render: (_text: string, record: ConditionItem) => {
        const type = record.content?.schemas[0]?.type || '';

        return <span>{type ? TYPE[type - 1] : ''}</span>;
      },
    },
    {
      title: global.actions,
      dataIndex: '',
      key: '',
      width: 100,
      render: (_text: string, record: ConditionItem) => (
        <>
          {group === ScopeEnum.project ? (
            <>
              <Button
                size="small"
                shape="circle"
                icon={<EditOutlined />}
                onClick={() => openDrawer(true, record, 'edit')}
              />
              <Divider type="vertical" />
              <Popconfirm
                title={`${global.deleteMsg}${record.name}?`}
                okText={global.yes}
                cancelText={global.no}
                onConfirm={() => handleConditionDelete(record.id)}>
                <Button size="small" shape="circle" icon={<DeleteOutlined />} />
              </Popconfirm>
            </>
          ) : (
            <Button
              size="small"
              shape="circle"
              icon={<EyeOutlined />}
              onClick={() => openDrawer(true, record, 'view')}
            />
          )}
        </>
      ),
    },
  ];

  const handleGroupChange = useCallback((value) => {
    setGroup(value);
  }, []);

  return (
    <>
      <Drawer
        title={<DrawerHeader>{global.condition}</DrawerHeader>}
        placement="left"
        visible
        destroyOnClose
        mask={false}
        closable={false}
        extra={<RightCloseIcon onClose={onClose} />}
        width={410}
        getContainer={container}
        onClose={onClose}
        headerStyle={{ padding: 16 }}
        bodyStyle={{ padding: 16 }}
        style={{
          position: 'absolute',
          top: 0,
          left: 0,
          zIndex: 50,
          boxShadow: 'rgba(0, 0, 0, 0.1) 4px 4px 4px 2px',
          overflowY: 'auto',
          transition: 'none',
        }}>
        <Toolbar>
          <ScopeSelect scope={group} onScopeChange={handleGroupChange} />
          {group === ScopeEnum.project && (
            <Button
              type="primary"
              size="small"
              icon={<PlusOutlined />}
              onClick={() => openDrawer(true, undefined, 'new')}>
              {conditionI18n.add}
            </Button>
          )}
        </Toolbar>
        <Spin spinning={fetching}>
          <Table
            columns={columns}
            bordered={false}
            rowKey={(record: ConditionItem) => record.id.toString()}
            pagination={false}
            dataSource={list}
            size="small"
          />
          <Pagination
            size="small"
            pageSize={PAGE_SIZE}
            current={pageNum}
            total={total}
            onChange={changeOffset}
          />
        </Spin>
      </Drawer>
      <EditDrawer applicationId={applicationId} folderId={folderId} />
    </>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(Component);
