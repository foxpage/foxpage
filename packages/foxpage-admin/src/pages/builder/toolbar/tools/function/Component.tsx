import React, { useCallback, useContext, useEffect, useState } from 'react';
import { connect } from 'react-redux';

import { DeleteOutlined, EditOutlined, EyeOutlined, PlusOutlined } from '@ant-design/icons';
import { Button, Divider, Drawer, message, Popconfirm, Spin, Table } from 'antd';
import { DrawerProps } from 'antd/lib/drawer';
import { RootState } from 'typesafe-actions';

import { Pagination } from '@/components/common/index';
import { ScopeEnum } from '@/constants/index';
import RightCloseIcon from '@/pages/builder/component/close';
import ScopeSelect from '@/pages/components/common/ScopeSelect';
import GlobalContext from '@/pages/GlobalContext';
import * as ACTIONS from '@/store/actions/builder/function';
import { FuncItem } from '@/types/application/function';

import BuilderContext from '../../../BuilderContext';
import { DrawerHeader, Toolbar } from '../common/CommonStyles';

import EditDrawer from './EditDrawer';

const PAGE_SIZE = 10;

const mapStateToProps = (state: RootState) => ({
  applicationId: state.builder.page.applicationId,
  folderId: state.builder.page.folderId,
  pageNum: state.builder.fn.pageNum,
  total: state.builder.fn.total,
  fetching: state.builder.fn.fetching,
  list: state.builder.fn.list,
});

const mapDispatchToProps = {
  changeOffset: ACTIONS.changeOffset,
  openDrawer: ACTIONS.updateFunctionDrawerVisible,
  fetchList: ACTIONS.fetchList,
  deleteCondition: ACTIONS.deleteFunction,
  clearAll: ACTIONS.clearAll,
};

interface DProps extends Omit<DrawerProps, 'onClose'> {
  onClose?: () => void;
}

type IProps = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps & DProps;

function Component(props: IProps) {
  const {
    applicationId,
    folderId,
    onClose,
    pageNum,
    total,
    fetching,
    list,
    changeOffset,
    openDrawer,
    fetchList,
    deleteCondition,
    clearAll,
  } = props;
  const { container } = useContext(BuilderContext);
  const [group, setGroup] = useState<ScopeEnum>(ScopeEnum.project);
  const { locale } = useContext(GlobalContext);
  const { global, function: functionI18n } = locale.business;

  useEffect(() => {
    return () => {
      clearAll();
    };
  }, []);

  useEffect(() => {
    fetchFunctionList();
  }, [applicationId, folderId, group, pageNum, fetchList]);

  const fetchFunctionList = () => {
    if (applicationId) {
      const params =
        group === ScopeEnum.project && !!folderId
          ? {
              applicationId,
              folderId,
              page: pageNum,
              size: PAGE_SIZE,
            }
          : {
              applicationId,
              type: 'live',
              page: pageNum,
              size: PAGE_SIZE,
            };

      fetchList(params);
    }
  };

  const handleConditionDelete = useCallback(
    (id) => {
      if (id && applicationId) {
        deleteCondition({
          applicationId,
          id,
          status: true,
        });
      } else {
        message.warning(global.deleteFailMsg);
      }
    },
    [applicationId, folderId, deleteCondition],
  );

  const columns = [
    {
      title: global.nameLabel,
      dataIndex: 'name',
      key: 'name',
    },
    {
      title: global.type,
      dataIndex: 'type',
      key: 'type',
      render: (_text: string, record: FuncItem) => {
        const type = record.content?.schemas[0]?.props?.async || '';

        return <span>{type ? 'async' : 'sync'}</span>;
      },
    },
    {
      title: global.actions,
      dataIndex: '',
      key: '',
      width: 100,
      render: (_text: string, record: FuncItem) => (
        <>
          {group === ScopeEnum.project ? (
            <>
              <Button
                size="small"
                shape="circle"
                icon={<EditOutlined />}
                onClick={() => openDrawer(true, record, 'edit')}
              />
              <Divider type="vertical" />
              <Popconfirm
                title={`${global.deleteMsg}${record.name}?`}
                okText="Yes"
                cancelText="No"
                onConfirm={() => handleConditionDelete(record.id)}>
                <Button size="small" shape="circle" icon={<DeleteOutlined />} />
              </Popconfirm>
            </>
          ) : (
            <Button
              size="small"
              shape="circle"
              icon={<EyeOutlined />}
              onClick={() => openDrawer(true, record, 'view')}
            />
          )}
        </>
      ),
    },
  ];

  const handleGroupChange = useCallback((group) => {
    setGroup(group);
  }, []);

  return (
    <>
      <Drawer
        title={<DrawerHeader>{functionI18n.name}</DrawerHeader>}
        placement="left"
        visible
        destroyOnClose
        mask={false}
        closable={false}
        extra={<RightCloseIcon onClose={onClose} />}
        width={410}
        getContainer={container}
        onClose={onClose}
        headerStyle={{ padding: 16 }}
        bodyStyle={{ padding: 16 }}
        style={{
          position: 'absolute',
          top: 0,
          left: 0,
          zIndex: 50,
          boxShadow: 'rgba(0, 0, 0, 0.1) 4px 4px 4px 2px',
          overflowY: 'auto',
          transition: 'none',
        }}>
        <Toolbar>
          <ScopeSelect scope={group} onScopeChange={handleGroupChange} />
          {group === ScopeEnum.project && (
            <Button
              type="primary"
              size="small"
              icon={<PlusOutlined />}
              onClick={() => openDrawer(true, undefined, 'new')}>
              {functionI18n.add}
            </Button>
          )}
        </Toolbar>
        <Spin spinning={fetching}>
          <Table
            columns={columns}
            bordered={false}
            rowKey={(record: FuncItem) => record.id.toString()}
            pagination={false}
            dataSource={list}
            size="small"
          />
          <Pagination
            size="small"
            pageSize={PAGE_SIZE}
            current={pageNum}
            total={total}
            onChange={changeOffset}
          />
        </Spin>
      </Drawer>
      <EditDrawer onSuccess={fetchFunctionList} />
    </>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(Component);
