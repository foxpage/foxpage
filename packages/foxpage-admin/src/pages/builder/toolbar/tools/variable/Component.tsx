import React, { useContext, useEffect, useState } from 'react';
import { connect } from 'react-redux';

import { EditOutlined, EyeOutlined, PlusOutlined } from '@ant-design/icons';
import { Button, Drawer, Popconfirm, Table } from 'antd';
import { DrawerProps } from 'antd/lib/drawer';
import { RootState } from 'typesafe-actions';

import * as ACTIONS from '@/actions/builder/variable';
import { ScopeEnum } from '@/constants/index';
import RightCloseIcon from '@/pages/builder/component/close';
import DeleteButton from '@/pages/common/DeleteButton';
import ScopeSelect from '@/pages/components/common/ScopeSelect';
import GlobalContext from '@/pages/GlobalContext';
import VariableType from '@/types/application/variable.d';

import BuilderContext from '../../../BuilderContext';
import { DrawerHeader } from '../common/CommonStyles';

import EditDrawer from './EditDrawer';

const mapStateToProps = (store: RootState) => ({
  loading: store.builder.variable.loading,
  variables: store.builder.variable.variables,
  storeApplicationId: store.builder.page.applicationId,
  storeFolderId: store.builder.page.folderId,
});

const mapDispatchToProps = {
  getVariables: ACTIONS.getVariables,
  getVariableBuildVersion: ACTIONS.getVariableBuilderVersion,
  openEditDrawer: ACTIONS.updateVariableEditDrawerOpen,
  deleteVariables: ACTIONS.deleteVariable,
  save: ACTIONS.saveVariable,
};

interface DProps extends Omit<DrawerProps, 'onClose'> {
  onClose?: () => void;
}

type Type = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps & DProps;

const Variable: React.FC<Type> = (props) => {
  const {
    loading,
    variables,
    storeApplicationId,
    storeFolderId,
    save,
    onClose,
    openEditDrawer,
    getVariables,
    getVariableBuildVersion,
    deleteVariables,
  } = props;
  const { container } = useContext(BuilderContext);
  const [group, setGroup] = useState<ScopeEnum>(ScopeEnum.project);
  const { locale } = useContext(GlobalContext);
  const { global, variable } = locale.business;

  useEffect(() => {
    setGroup(ScopeEnum.project);
    if (storeApplicationId && storeFolderId) {
      getVariables(storeFolderId);
    }
  }, [storeApplicationId, storeFolderId]);

  const columns = [
    {
      title: global.nameLabel,
      dataIndex: 'name',
      key: 'name',
    },
    {
      title: global.type,
      dataIndex: 'type',
      key: 'type',
      render: (_text: string, record: VariableType) => {
        const schemas = record.content?.schemas;
        return <React.Fragment>{schemas.length > 0 ? schemas[0].type : '--'}</React.Fragment>;
      },
    },
    {
      title: global.actions,
      key: 'updateTime',
      width: 72,
      render: (_text: string, record: VariableType) => {
        return (
          <React.Fragment>
            {group === ScopeEnum.project && (
              <React.Fragment>
                <Button
                  type="default"
                  size="small"
                  shape="circle"
                  title={global.edit}
                  onClick={() => {
                    openEditDrawer(record);
                    getVariableBuildVersion(record, storeApplicationId);
                  }}>
                  <EditOutlined />
                </Button>
                <Popconfirm
                  title={`${global.deleteMsg}${record.name}`}
                  onConfirm={() => {
                    deleteVariables({
                      applicationId: storeApplicationId,
                      fileId: record.id,
                      folderId: storeFolderId,
                    });
                  }}
                  okText={global.yes}
                  cancelText={global.no}>
                  <DeleteButton
                    type="default"
                    size="small"
                    shape="circle"
                    title={global.remove}
                    style={{ marginLeft: 4, marginRight: 0 }}
                  />
                </Popconfirm>
              </React.Fragment>
            )}
            {group === 'application' && (
              <React.Fragment>
                <Button
                  type="default"
                  size="small"
                  shape="circle"
                  title="View"
                  onClick={() => {
                    openEditDrawer(record);
                    getVariableBuildVersion(record, storeApplicationId);
                  }}
                  style={{ marginLeft: 8 }}>
                  <EyeOutlined />
                </Button>
              </React.Fragment>
            )}
          </React.Fragment>
        );
      },
    },
  ];
  return (
    <React.Fragment>
      <Drawer
        title={<DrawerHeader>{variable.title}</DrawerHeader>}
        placement="left"
        visible
        destroyOnClose
        mask={false}
        closable={false}
        extra={<RightCloseIcon onClose={onClose} />}
        width={410}
        getContainer={container}
        onClose={onClose}
        headerStyle={{ padding: 16 }}
        bodyStyle={{ padding: 16 }}
        style={{
          position: 'absolute',
          top: 0,
          left: 0,
          zIndex: 50,
          boxShadow: 'rgba(0, 0, 0, 0.1) 4px 4px 4px 2px',
          overflowY: 'auto',
          transition: 'none',
        }}>
        <div style={{ marginBottom: 12, display: 'flex', justifyContent: 'space-between' }}>
          <ScopeSelect
            scope={group}
            onScopeChange={(group) => {
              setGroup(group);
              if (group === ScopeEnum.project) {
                getVariables(storeFolderId);
              } else {
                getVariables(undefined, 'live');
              }
            }}
          />
          {group === ScopeEnum.project && (
            <Button
              size="small"
              type="primary"
              onClick={() => {
                openEditDrawer(true);
              }}>
              <PlusOutlined /> {variable.add}
            </Button>
          )}
        </div>
        <Table
          rowKey="id"
          loading={loading}
          dataSource={variables}
          columns={columns}
          pagination={false}
          size="small"
        />
      </Drawer>
      <EditDrawer
        applicationId={storeApplicationId}
        folderId={storeFolderId}
        onSave={() => {
          save({ folderId: storeFolderId, applicationId: storeApplicationId });
        }}
      />
    </React.Fragment>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(Variable);
