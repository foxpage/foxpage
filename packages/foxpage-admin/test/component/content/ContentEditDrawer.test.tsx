import React from 'react';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import Component from '../../../src/pages/components/business/content/ContentEditDrawer';
import { locales } from '../../global.data';
import { content } from '../../mock/content.data';

const mockStore = configureStore([]);

describe('component-content-edit', () => {
  const onClose = jest.fn();
  const onSave = jest.fn();
  const props = {
    open: true,
    locales,
    content,
    onSave,
    onClose,
  };
  it('match snapshot', async () => {

    const wrapper = mount(
      <Provider store={mockStore({})}>
        <Component {...props} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
