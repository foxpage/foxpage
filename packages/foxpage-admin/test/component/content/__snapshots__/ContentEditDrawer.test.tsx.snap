// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-content-edit match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ContentEditDrawer
    content={
      Object {
        "creator": Object {
          "account": "fox-user",
          "email": "",
          "id": "user-dvbfW1qsl8qkz39",
        },
        "fileId": "file_aPa0mIILqkc8sBO",
        "id": "cont_cjS8BJyOkyVzBY8",
        "tags": Array [
          Object {
            "locale": "en-US",
          },
        ],
        "title": "promo",
        "urls": Array [
          "",
        ],
      }
    }
    locales={
      Array [
        "en-US",
      ]
    }
    onClose={[MockFunction]}
    onSave={[MockFunction]}
    open={true}
  >
    <OperationDrawer
      actions={
        <Button
          onClick={[Function]}
          type="primary"
        >
          Apply
        </Button>
      }
      anchor="right"
      canExpend={false}
      destroyOnClose={true}
      maskClosable={true}
      onClose={[MockFunction]}
      open={true}
      title="Edit"
      width={480}
    >
      <Styled(DrawerWrapper)
        closable={false}
        destroyOnClose={true}
        key="xwR-N6hyhHS"
        maskClosable={true}
        onClose={[MockFunction]}
        placement="right"
        visible={false}
        width={480}
      >
        <DrawerWrapper
          className="sc-dkPtRN cfcQnA"
          closable={false}
          destroyOnClose={true}
          maskClosable={true}
          onClose={[MockFunction]}
          placement="right"
          visible={false}
          width={480}
        >
          <Drawer
            className="sc-dkPtRN cfcQnA"
            closable={false}
            destroyOnClose={true}
            maskClosable={true}
            onClose={[MockFunction]}
            placement="right"
            prefixCls="ant-drawer"
            visible={false}
            width={480}
          >
            <DrawerWrapper
              afterVisibleChange={[Function]}
              autoFocus={true}
              className="sc-dkPtRN cfcQnA"
              defaultOpen={false}
              duration=".3s"
              ease="cubic-bezier(0.78, 0.14, 0.15, 0.86)"
              forceRender={false}
              getContainer="body"
              handler={false}
              keyboard={true}
              level={null}
              maskClosable={true}
              maskStyle={Object {}}
              onChange={[Function]}
              onClose={[MockFunction]}
              open={false}
              placement="right"
              prefixCls="ant-drawer"
              showMask={true}
              style={
                Object {
                  "transform": undefined,
                  "zIndex": undefined,
                }
              }
              width={480}
              wrapperClassName=""
            >
              <PortalWrapper
                forceRender={false}
                getContainer="body"
                visible={false}
                wrapperClassName=""
              />
            </DrawerWrapper>
          </Drawer>
        </DrawerWrapper>
      </Styled(DrawerWrapper)>
    </OperationDrawer>
  </ContentEditDrawer>
</Provider>
`;
